src/components/Select/Select.js

```diff
@@ -23,37 +24,41 @@ export const Select = ({
   formTouched,
   labelCopy,
   name,
+  initialValue,
+  options,
   ...rest
 }) => {
-  const onChangeHandler = (event) => {
-    updateSelectedValue(event.value)
-
-    if (onChange) {
-      onChange(event)
-    }
-  }
-
   const resolvedValidator = validator ? validator : () => ''
+  const [selectedOption, updateSelectedOption] = useState(initialValue)
+  const [touched, setTouched] = useState(false)
   const [getError, setError, , validate] = useErrorMessage(resolvedValidator)
-  const [selectedValue, updateSelectedValue] = useState(undefined)
+  const [doValidation] = useInputValidation({
+    validate,
+    setError,
+    formChangeHandler,
+  })
 
-  const validationSelect = () => {
-    const errorMessage = validate(selectedValue)
-    setError(errorMessage)
-    if (formChangeHandler) {
-      formChangeHandler(selectedValue, errorMessage)
-    }
-  }
+  const getSelectedValue = () =>
+    selectedOption ? selectedOption.value : undefined
 
   useEffect(() => {
-    const isSelectedValue = typeof selectedValue !== 'undefined'
-    if (isSelectedValue) {
-      validationSelect()
+    if (!!formChangeHandler && touched) {
```

Comment:
not sure if the `useInputValidation` is doing that check for us already in which case this'd be duplication

Reply:
yeah, you're right! I removed the formChangeHandler, but I needed to keep the touched check because otherwise the form would be 'interacted with' immediately.